carla_ego_vehicle/src/carla_ego_vehicle/carla_ego_vehicle.py

```diff
@@ -16,52 +16,86 @@
 
 /initialpose might be published via RVIZ '2D Pose Estimate" button.
 """
-
-from abc import abstractmethod
-
+import json
+import math
 import os
 import random
-import math
-import json
-import rospy
-from tf.transformations import euler_from_quaternion, quaternion_from_euler
+from abc import abstractmethod
+
+ROS_VERSION = int(os.environ['ROS_VERSION'])
+
+if ROS_VERSION == 1:
+    import rospy
+    from tf.transformations import euler_from_quaternion, quaternion_from_euler
+    from ros_compatibility import CompatibleNode
+
+
+elif ROS_VERSION == 2:
+    import sys
+    import os
+    print(os.getcwd())
+    # TODO: fix setup.py to easily import CompatibleNode (as in ROS1)
+    sys.path.append(os.getcwd() + '/install/ros_compatibility/lib/python3.6/site-packages/src/ros_compatibility')
+    import rclpy
+    from rclpy.node import Node
+    from rclpy import executors
+    from ament_index_python.packages import get_package_share_directory
+    from transformations.transformations import euler_from_quaternion, quaternion_from_euler
```

Comment:
https://pypi.org/project/transformations/ -> seems that this could give some problems. The feeling I get from the resource quote: 

> Transformations.py is no longer actively developed and has a few known issues and numerical instabilities. The module is mostly superseded by other modules for 3D transformations and quaternions:

Maybe it will be worth it to take a look at `tf2` to see if it provides the functionalities we need. The official port is in: https://github.com/ros2/geometry2

Reply:
`transformations` module is what ros1 integrated in their `tf` package. `tf2` currently doesn't provide the functionality of rotation conversion (in `tf2` tutorials they use `tf.transformations` for that). We will probably use `scipy` for rotation conversion later on.